netlink/mocknetlink.go

```diff
@@ -0,0 +1,92 @@
+package netlink
+
+import (
+	"errors"
+	"fmt"
+	"net"
+)
+
+var errorMockNetlink = errors.New("Mock Netlink Error")
+
+func newErrorMockNetlink(errStr string) error {
+	return fmt.Errorf("ErrorMockNetlink %w : %s", errorMockNetlink, errStr)
+}
+
+type MockNetlink struct {
+	returnError bool
+	errorString string
+}
+
+func NewMockNetlink(returnError bool, errorString string) MockNetlink {
+	return MockNetlink{
+		returnError: returnError,
+		errorString: errorString,
+	}
+}
+
+func (f *MockNetlink) error() error {
+	if f.returnError {
+		return newErrorMockNetlink(f.errorString)
+	}
+	return nil
+}
+
+func (f *MockNetlink) AddLink(Link) error {
+	return f.error()
+}
+
+func (f *MockNetlink) DeleteLink(string) error {
+	return f.error()
```

Comment:
How is the interface that this mock implements used? The problem with having a single flag for all returns is that it restricts different permutations on the behavior this type displays. As a silly example, if a client has 
```go
func foo(m NetlinkInterface) error {
    if err := m.AddLink(...); err != nil {
        return err
    }

    return m.DeleteLink(...)
}
```

Then I wouldn't be able to cover the last line with a test.

Reply:
I am too worried about how the mock works underneath, we can change it as we start writing tests, this PR is to build the infra. In consecutive PRs for adding tests with Mocknetlink we can improve it based on requirement